api/imagebuilder/janitor.go

```diff
@@ -0,0 +1,54 @@
+package imagebuilder
+
+import (
+	"time"
+
+	"github.com/gojek/merlin/cluster"
+	"github.com/gojek/merlin/log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Janitor cleans the finished image building jobs.
+type Janitor struct {
+	cc  cluster.Controller
```

Comment:
I think it's cleaner to just directly depend to kubernetes API here

Reply:
I think it's better to refactor our cluster.Controller too. For now, I'm going to keep it as is. I'll keep your idea in mind for future changes.